config/environments/test.rb

```diff
@@ -14,8 +14,8 @@
   config.eager_load = false
 
   # Configure static file server for tests with Cache-Control for performance.
-  config.serve_static_files   = true
-  config.static_cache_control = 'public, max-age=3600'
+  config.public_file_server.enabled = true
+  config.public_file_server.headers = { 'Cache-Control' => 'public, max-age=3600' }
```

Comment:
Maybe you'd better disable caching headers here? It will slow your tests down a little bit, but make their results more reliable. Smart tradeoff, in my view.